### ISSUE TITLE
Dat 2 Support

### ISSUE THREAD

**Issue by RangerMauve** [datproject/sdk#36]
- [x] Add corestore for replication
  - [x] Hyperdrive 10 API with promise wrapper
  - [x] Hypercore API with promise wrapper
  - [x] Use hyperswarm-web for web
  - [x] Use [sodium-javascript-plus](https://github.com/geut/sodium-javascript-plus) to use new hypercore on web
  - [x] Add `destroy` method to hyperdrive (pending on [PR](https://github.com/mafintosh/hyperdrive/pull/271))
  - [x] Add way to control `discoveryKey` / `lookup` / `announce` swarm opts.
  - [x] Docs
  - [x] Release

This is currently blocked on hypercore having support for browsers / API stabilization of corestore and hyperdrive

**Comment by RangerMauve** [comment]
Working on this in this branch: https://github.com/datproject/sdk/tree/dat2-take-two

Just waiting for the destroyData method to be merged into Hypercore and Hyperdrive to release. 😁

**Comment by RangerMauve** [comment]
This is pretty much ready. Gonna release once the hypercore protocol team does their release and the code can be considered stable. (this week?)

**Comment by RangerMauve** [comment]
Woot. Hypercore seems to be done with all their breaking changes.

Seems there's an issue with getting it to work on the web still: https://github.com/geut/sodium-javascript-plus/issues/2

Might wait for that to be figured out or I'll release it as is for node.js use.

**Comment by tinchoz49** [comment]
Hey @RangerMauve! I tested the sdk with the latest (v0.0.6) sodium-javascript-plus and it seems to be working. I got to replicate a feed between the browser and node.

**Comment by RangerMauve** [comment]
This is all out now!
